consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/validation/ProposalPayloadValidator.java

```diff
@@ -89,4 +104,58 @@ private boolean validateBlock(final Block block) {
 
     return true;
   }
+
+  /*
+    Validate CMS in block header (only for PKI-mode)
+  */
+  private boolean validateCms(final Block block) {
+    final PkiQbftContext pkiQbftContext;
+    try {
+      pkiQbftContext = protocolContext.getConsensusState(PkiQbftContext.class);
+    } catch (final ClassCastException e) {
+      // TODO-lucas Do we have a better way of doing this?
+      // Not running on PKI-enabled mode
+      return true;
+    }
```

Comment:
Perhaps pass the PkiBlockCreationConfiguration as an optional constructor argument to the MessageValidatorFactory you could then use that to test if PKI is enabled without the need to cast or use the context.